### DISCUSSION TITLE
How can a mutation trigger invalidating a specific query in document caching

### DISCUSSION THREAD

**Discussion by Biboswan** [urql-graphql/urql#2302] [Q&A] [answered]
I'm aware of the pattern Eg `mutationAction({... }, { additionalTypenames: ['Feed'] })`

But what I want is something like `mutationAction({... }, { additionalQueries: ['ABC_QUERY'] })`
Basically, I want to refetch some queries in response to a mutation.

**Comment by JoviDeCroock** [comment]
That's not how document-caching works atm, we currently invalidate and tracked based on the types contained within an entity. In our normalized cache there is more fine-grained control over these

**Comment by kitten** [comment, marked as answer]
You'd have to modify the document `cacheExchange` yourself if you wanted this.
We basically don't implement this out of principle. Generally speaking, operation names are arbitrary. They can change any time and may just as much be comments or phrases that don't mean much. If you invalidate by them or take them into account in caching, you're likely just targeting a very specific query in a whole different part of the app.
